assets/js/googlesitekit/datastore/site/info.js

```diff
@@ -162,6 +162,7 @@ export const reducer = ( state, { payload, type } ) => {
 				pluginBasename,
 				productBasePaths,
 				keyMetricsSetupCompleted,
+				keyMetricsSetupCompletedByUserID,
```

Comment:
It seems unnecessary to keep these as separate bits of state. If the setup was completed, we should have an associated user ID, so that part can be deduced from the same `keyMetricsSetupCompletedByUserID` state. It could keep its own selector if we want that as a more expressive selector, but it just seems unnecessary to keep both bits of state. I thought we were changing the existing option to only store the user ID as well, so perhaps this was just missed?

Reply:
@aaemnnosttv This is saving the user ID instead of boolean, but that option is still returning boolean and we have that new key which pulls the user ID, this all according to IB

Reply:
We shouldn't be _saving_ both the boolean and the number, just the number, as written in the IB:

![CleanShot 2023-10-04 at 10 14 30](https://github.com/google/site-kit-wp/assets/90871/b4c0cac6-fd1c-4d71-b87a-1780db7e0933)

I agree that we should keep the boolean selector around, but only the user ID should be saved, and as @aaemnnosttv said: if it isn't truthy then we know the setup was not yet completed 👍🏻

Reply:
@aaemnnosttv I started changing this to exclude that new ID selector, but the issue I encountered on is that we have this code https://github.com/google/site-kit-wp/blob/19664dc93b09e60b57fb1a6645be730fe09e4145/assets/js/googlesitekit/datastore/user/key-metrics.js#L140-L142
Which is syncing this to boolean. And if using only one selector for this, it will override it to true, and I can't use current user ID here, as user ID is saved only of the user who did initial setup. It will introduce unwanted side effects, it seems better to stay with IB on this one.

Reply:
@zutigrm it sounds like you may be misunderstanding the suggestion – I'm not suggesting that we only use one selector – having separate selectors is totally fine. My point was that we don't need to pass and store data separately in state for this. The only state we need on the client is the user ID that completed the setup. If this is a number greater than 0, then that means setup has also been completed. In some cases we only care about the boolean so in this case having that as a separate selector can be useful. So keeping both selectors as described in the IB is fine. Keeping the setup and ID as separate state is unnecessary and can lead to contradictory states where, for example a user ID is set but "is completed" is false, which doesn't make sense. 

Regarding the example above, I'm not sure we need to synchronize the completed state for the user who is completing the setup (https://github.com/google/site-kit-wp/issues/7346#issuecomment-1736605121), but even if we did, we'd simply need to update this call to receive the ID instead (this would be the current user in this case I believe) rather than setting a boolean – the effect would be the same.

Currently the IB is calling to use the same state for both bits of data it's passing to the client anyways, I'm only suggesting that we do the same but only on the client instead using the same state, and only passing the ID from the server.

Does that make sense?

Reply:
Resolving as we clarified this earlier on our call.